drivers/SmartThings/zigbee-window-treatment/src/mic/init.lua

```diff
@@ -0,0 +1,147 @@
+  local invert_value = device.preferences.reverse
+  local invert_cluster_cmd = cluster_base.write_manufacturer_specific_attribute(device, INVERT_CLUSTER,
+      INVERT_CLUSTER_ATTRIBUTE, 0x0000, data_types.Boolean, invert_value)
+  device:send(invert_cluster_cmd)
+end
+
+local function info_changed(driver, device, event, args)
+  if device.preferences ~= nil and device.preferences.reverse ~= args.old_st_store.preferences.reverse then
+    invert_preference_handler(device)
+  end
+end
+
+local set_window_shade_level = function(level)
+  return function(driver, device, cmd)
+    device:send_to_component(cmd.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+  end
+end
+
+local function current_position_attr_handler(driver, device, value, zb_rx)
+
+  local level = value.value
+  local current_level = device:get_latest_state("main", capabilities.windowShadeLevel.ID,
+      capabilities.windowShadeLevel.shadeLevel.NAME)
+  local windowShade = capabilities.windowShade.windowShade
+  if level == 0 then
+    device:emit_event(windowShade.closed())
+    device:emit_event(capabilities.windowShadeLevel.shadeLevel(0))
+  elseif level == 100 then
+    device:emit_event(windowShade.open())
+    device:emit_event(capabilities.windowShadeLevel.shadeLevel(100))
+  else
+    if current_level ~= level or current_level == nil then
+      current_level = current_level or 0
+      device:emit_event(capabilities.windowShadeLevel.shadeLevel(level))
+      local event = nil
+      event = current_level < level and windowShade.partially_open()
+      if current_level ~= level then
+        device:emit_event(event)
+      end
+    end
+    local current_level = device:get_latest_state("main", capabilities.windowShadeLevel.ID,
+        capabilities.windowShadeLevel.shadeLevel.NAME)
+    if current_level == 0 then
+      device:emit_event(windowShade.closed())
+    elseif current_level == 100 then
+      device:emit_event(windowShade.open())
+    else
+      device:emit_event(windowShade.partially_open())
+    end
+
+  end
+end
+
+local function window_shade_level_cmd(driver, device, command)
+  local level = utils.clamp_value(command.args.shadeLevel, 0, 100)
+  device:send_to_component(command.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+end
+
+local function window_shade_preset_cmd(driver, device, command)
+  if device.preferences ~= nil and device.preferences.presetPosition ~= nil then
+    device:send_to_component(command.component, WindowCovering.server.commands
+        .GoToLiftPercentage(device, device.preferences.presetPosition))
+  end
+end
+
+local do_refresh = function(self, device)
+  device:send(WindowCovering.attributes.CurrentPositionLiftPercentage:read(device))
+  local invert_cluster_read = cluster_base.read_manufacturer_specific_attribute(device, INVERT_CLUSTER,
+      INVERT_CLUSTER_ATTRIBUTE, 0x0000)
+  device:send(invert_cluster_read)
```

Comment:
right, so this read is sent but the response is never handled. There's no reason to send a read if you don't care about the response

if you want to handle the response, you would add an `attr` handler for `INVERT_CLUSTER`'s `INVERT_CLUSTER_ATTRIBUTE`. See: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-humidity-sensor/src/centralite-sensor/init.lua#L87